src/net/ipv4.c

```diff
@@ -844,6 +845,36 @@ static int ipv4_gratuitous_arp ( struct net_device *netdev,
 	return 0;
 }
 
+/**
+ * Convert netmask from CIDR notation to the common x.x.x.x notation
+ */
+static inline uint8_t cidr_to_mask ( uint8_t cidr, char * buf ) {
+	uint8_t addr_len = 0;
+	uint8_t n_octet = ( cidr % 8 == 0 ? 255 :
+			( 255 - ( ( 256 >> ( cidr % 8 ) ) - 1 ) ) );
+
+	if (cidr > 24 && cidr <= 32) {
+		sprintf(buf, "255.255.255.%d", n_octet);
+		addr_len = 4;
+	} else if (cidr > 16) {
+		sprintf(buf, "255.255.%d.0", n_octet);
+		addr_len = 3;
+	} else if (cidr > 8) {
+		sprintf(buf, "255.%d.0.0", n_octet);
+		addr_len = 2;
+	} else if (cidr > 0) {
+		sprintf(buf, "%d.0.0.0", n_octet);
+		addr_len = 1;
+	} else if (cidr == 0) {
+		sprintf(buf, "0.0.0.0");
+		addr_len = 0;
+	} else {
+		DBGC(cidr, "Unknown netmask descriptor /%d.\n", cidr);
```

Comment:
I don't quite get what's the point of having an individual format string for each full byte. Isn't converting it to a netmask already sufficient ?

```python
def cidr_to_netmask(cidr):
     return tuple(0xFF & ~((1 << (8 - min(8, cidr - i * 8))) - 1) for i in range(4))
```